### PR TITLE
#428 [SA] when finalStates are dumped in contingencies, reorder themin a clean folder

### PR DESCRIPTION
**Pull request by barondim**
**Checklist before requesting a review**

*use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes*

- [ ] unit tests and non regression tests were added (update of algorithms)
- [ ] main documentation was updated (update of input/output file, update of algorithms)
- [x] the corresponding milestone was added in the ticket and in this PR
- [ ] if this PR modifies a dictionary: the corresponding french dictionary was updated

### CODE REVIEW

**Comment by barondim** [comment]
to be merged with https://github.com/dynawo/dynawo/pull/3459

File: sources/Common/DYNSimulationResult.h [hunk 1]
```diff
@@ -127,6 +127,10 @@ class SimulationResult {
    */
   std::string getLostEquipementsStreamStr() const;
 
+  std::stringstream& getOutputIIDMStream();
```
Comment (rosiereflo) [inline sources/Common/DYNSimulationResult.h:130]: Missing doxygen

File: sources/Launcher/DYNRobustnessAnalysisLauncher.cpp [hunk 2]
```diff
@@ -380,6 +380,7 @@ RobustnessAnalysisLauncher::simulate(const boost::shared_ptr<DYN::Simulation>& s
       simulation->terminate();
       result.setSuccess(true);
       result.setStatus(CONVERGENCE_STATUS);
+      simulation->dumpIIDMFile(result.getOutputIIDMStream());
```
Comment (rosiereflo) [inline sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:383]: Why is this done here and not in the same place than the others? Also the iidm should not be written if finalState_.iidmFile_  is false on dynawo side (in this case this stream should remain empty)

File: sources/Launcher/DYNRobustnessAnalysisLauncher.cpp [hunk 3]
```diff
@@ -423,6 +423,7 @@ RobustnessAnalysisLauncher::simulate(const boost::shared_ptr<DYN::Simulation>& s
     simulation->printTimeline(result.getTimelineStream());
     simulation->printConstraints(result.getConstraintsStream());
     simulation->printLostEquipments(result.getLostEquipementsStream());
+    simulation->dumpIIDMFile(result.getOutputIIDMStream());
```
[the remark this answers is not in the document]
Comment (barondim) [reply sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:426]: If <dyn:finalState exportDumpFile="true" exportIIDMFile="true"/> is in the .jobs file, the outputIIDM.xml files wil be dumped.

[the remark this answers is not in the document]
Comment (barondim) [reply sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:426]: if the exportIIDMFile="false", outputIIDM.xml won't be dumped with the current version

[the remark this answers is not in the document]
Comment (rosiereflo) [reply sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:426]: where is the test? If I'm not mistaken n dynawo there is no test on the jobs options, the dumpIIDMFile dumps the file no matter what

[the remark this answers is not in the document]
Comment (barondim) [reply sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:426]: In DYNSimulation.cpp l.1470 :
`void
Simulation::dumpIIDMFile(std::stringstream& stream) const {
  if (data_ && finalState_.iidmFile_)
    data_->dumpToFile(stream);
}`

We dump only if `exportIIDMFile="false"`

[the remark this answers is not in the document]
Comment (rosiereflo) [reply sources/Launcher/DYNRobustnessAnalysisLauncher.cpp:426]: ok I was looking at the filename version of dumpIIDM which is not protected. Thanks

**Review by rosiereflo — APPROVED** [review]

**Pull request merged**